index.js

```diff
@@ -0,0 +1,151 @@
+let myLibrary = [];
+const setStorage = myLibrary =>
+  localStorage.setItem("myLibrary", JSON.stringify(myLibrary));
+const getStorage = myLibrary => {
+  if (myLibrary.length) {
+    setStorage(myLibrary);
+  } else if (localStorage.getItem("myLibrary")) {
+    const item = JSON.parse(localStorage.getItem("myLibrary"));
+    myLibrary = item.map(
+      ({ title, shortdescription, pages, author, status }) => {
+        const newBook = new Book(
+          title,
+          shortdescription,
+          pages,
+          author,
+          status
+        );
+        return newBook;
+      }
+    );
+  }
+  return myLibrary;
+};
+
+const addBookButton = document.querySelector("#add_book");
+addBookButton.addEventListener("click", addBookToLibrary);
+
+// all deleted items
+const recycler = [];
+
+if (storageAvailable()) {
+  if (localStorage.getItem("myLibrary")) {
+    myLibrary = getStorage(myLibrary);
+  }
+} else {
+  myLibrary = [];
+}
+
+function Book(title, shortdescription, pages, author) {
+  this.title = title;
+  this.shortdescription = shortdescription;
+  this.pages = pages;
+  this.author = author;
+  this.status = "available";
+  this.read = function() {
+    if (this.status === "available") {
+      this.status = "busy";
+    } else {
+      this.status = "available";
+    }
+    setStorage(myLibrary);
+    render();
+  };
+}
+
+const readBook = function(index) {
+  return myLibrary[index].read();
+};
+const removeBookFromLibrary = index => {
+  recycler.push(myLibrary.splice(index, 1));
+  setStorage(myLibrary);
+  render();
+};
+const changeBookState = ({ target }) => {
+  const { value, name } = target;
+  const index = Number(name);
+  switch (value) {
+    case "dispose":
+      removeBookFromLibrary(index);
+      break;
+    case "take to read":
+      readBook(index);
+      break;
+    default:
+      break;
+  }
+};
+
+function addBookToLibrary() {
+  const book = new Book();
+  let validEntry = true;
+  const bookDetails = document.querySelectorAll("input");
+  const getBookDetails = ({ name, value, validity: { valid = true } }) => {
+    if (!valid) {
+      validEntry = valid;
+    }
+    book[`${name}`] = value;
+  };
+
+  bookDetails.forEach(getBookDetails);
+  if (validEntry) {
```

Comment:
Would be nice to show the user some kind of feedback when the input is not valid.

Also after adding the book it would be preferable to clear the from.